File: 283-move-zeroes/283.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if not nums:
            return
        size = len(nums)
        cur = 0
        for i in range(size):
            if nums[i] != 0:
                nums[cur] = nums[i]
                cur += 1
        
        for i in range(cur, size):
            nums[i] = 0

if __name__ == '__main__':
    sol = Solution()
    nums = [0,1,0,3,12]
    sol.moveZeroes(nums)
    print(nums)

File: shoope/283-move-zeroes/283.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        if (nums.empty()) return;
        int size = nums.size();
        int cur = 0;
        for (int i = 0; i < size; ++i) {
            if (nums[i] != 0) {
                nums[cur++] = nums[i]; 
            }
        }
        while (cur < size) {
            nums[cur++] = 0;
        }
    }
};

int main(int argc, char** args) {
    Solution sol = Solution();
    vector<int> nums = {0, 1, 0, 3, 12};
    sol.moveZeroes(nums);
    for (auto a : nums) {
        cout << a << " ";
    }
    cout << endl;
    return 0;
}